meson.build

```diff
@@ -0,0 +1,624 @@
+                                 #include <keyutils.h>'''],
+        ['bpf',               '''#include <sys/syscall.h>
+                                 #include <unistd.h>'''],
+        ['statx',             '''#include <sys/types.h>
+                                 #include <sys/stat.h>
+                                 #include <unistd.h>'''],
+        ['pidfd_send_signal', '''#include <stdlib.h>
+                                 #include <unistd.h>
+                                 #include <signal.h>
+                                 #include <sys/wait.h>'''],
+        ['pidfd_open',        '''#include <stdlib.h>
+                                 #include <unistd.h>
+                                 #include <signal.h>
+                                 #include <sys/wait.h>'''],
+        ['execveat',          '''#include <unistd.h>'''],
+        ['close_range',       '''#include <unistd.h>'''],
+        ['mount_setattr',     '''#include <sys/mount.h>'''],
+        ['move_mount',        '''#include <sys/mount.h>'''],
+        ['open_tree',         '''#include <sys/mount.h>'''],
+        ['strlcpy',	      '''#include <string.h>'''],
+        ['strlcat',	      '''#include <string.h>'''],
+        ['sethostname',	      '''#include <unistd.h>'''],
+        ['faccessat',         '''#include <fcntl.h>
+                                 #include <unistd.h>'''],
+        ['unshare',	      '''#include <sched.h>'''],
+        ['prlimit',	      '''#include <sys/time.h>
+                               #include <sys/resource.h>'''],
+        ['prlimit64',	      '''#include <sys/time.h>
+                               #include <sys/resource.h>'''],
+]
+
+        have = cc.has_function(ident[0], prefix : ident[1], args : '-D_GNU_SOURCE')
+	if have == true
+		conf.set10('HAVE_' + ident[0].to_upper(), have)
+	endif
+endforeach
+
+threads = dependency('threads')
+libseccomp = cc.find_library('seccomp', required : false)
+if libseccomp.found()
+	conf.set10('HAVE_SECCOMP', libseccomp.found())
+
+	seccomp_headers = '''
+	#include <seccomp.h>
+	'''
+
+	if cc.has_function('seccomp_notify_fd', prefix : seccomp_headers, args : '-D_GNU_SOURCE', dependencies: libseccomp)
+	        conf.set10('HAVE_DECL_SECCOMP_NOTIFY_FD', true)
+	endif
+
+	if cc.has_function('seccomp_syscall_resolve_name_arch', prefix : seccomp_headers, args : '-D_GNU_SOURCE', dependencies: libseccomp)
+	        conf.set10('HAVE_DECL_SECCOMP_SYSCALL_RESOLVE_NAME_ARCH', true)
+	endif
+
+	foreach decl : [
+	                'scmp_filter_ctx',
+	                'struct seccomp_notif_sizes',
+	                'struct clone_args',
+	               ]
+
+	        # We get -1 if the size cannot be determined
+	        have = cc.sizeof(decl, prefix : seccomp_headers, args : '-D_GNU_SOURCE') > 0
+
+		if have == true
+			conf.set10('HAVE_' + decl.underscorify().to_upper(), have)
+		endif
+	endforeach
+endif
+
+libselinux = cc.find_library('selinux', required : false)
```

Comment:
Has a pkg-config file.

Reply:
Doesn't find it here with `dependency()` but doe find it with `find_library()`.

Reply:
```
Run-time dependency selinux found: NO (tried pkgconfig and cmake)
Run-time dependency apparmor found: NO (tried pkgconfig and cmake)
Run-time dependency openssl found: YES 1.1.1j
Run-time dependency libcap found: YES 2.44
```

Reply:
Ah ok, I see. I need to look for `libselinux` instead of `selinux`. Weird, I would've expected `find_library` to require `libselinux` not dependency.

Reply:
dependency() will look for the exact filename minus .pc, because that is the calling convention of pkg-config, `pkg-config --cflags --libs {name}`..

cc.find_library() will look for the library name minus the platform prefix (lib) and suffix (.so), because that is the calling convention of the compiler, `-l{name}`.